src/cli/load-compiler.ts

```diff
@@ -1,9 +1,14 @@
 import type { CompilerSystem } from '../declarations';
 
 export const loadCoreCompiler = async (sys: CompilerSystem): Promise<CoreCompiler> => {
-  await sys.dynamicImport(sys.getCompilerExecutingPath());
+  const compilerMod = await sys.dynamicImport!(sys.getCompilerExecutingPath());
 
-  return (globalThis as any).stencil;
+  if ((globalThis as any).stencil) {
+    return (globalThis as any).stencil;
+  } else {
+    (globalThis as any).stencil = compilerMod;
+    return compilerMod;
+  }
```

Comment:
Last question - for my own edification, why do we need this?

Reply:
this is basically the minimum change I was able to figure out that would keep this little bit of code working in both the rollup-based and esbuild-based setup. I think the `globalThis` stuff is a little hack to avoid importing things multiple times